service-base/src/main/java/org/eclipse/hono/service/AbstractApplication.java

```diff
@@ -50,12 +48,12 @@
     protected final Logger log = LoggerFactory.getLogger(getClass());
 
     private ObjectFactory<S> serviceFactory;
-    private C config;
+    private ApplicationConfigProperties config;
```

Comment:
That doesn't help outside of a Spring context. I also do not really understand where you see/suspect an inconsistency. Can you help me out with that?